components/eppp_link/eppp_uart.c

```diff
@@ -0,0 +1,213 @@
+/*
+ * SPDX-FileCopyrightText: 2025 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <string.h>
+#include <stdint.h>
+#include "sdkconfig.h"
+#include "esp_log.h"
+#include "esp_netif.h"
+#include "esp_check.h"
+#include "esp_event.h"
+#include "eppp_link.h"
+#include "eppp_transport.h"
+#include "driver/uart.h"
+
+#define TAG "eppp_uart"
+
+struct eppp_uart {
+    struct eppp_handle parent;
+    QueueHandle_t uart_event_queue;
+    uart_port_t uart_port;
+};
+
+#define UART_BUF_SIZE (1024)
+#define MAX_PAYLOAD (1500)
+#define HEADER_MAGIC (0x7E)
+#define PACKET_SIZE (MAX_PAYLOAD + 4)
+
+struct header {
+    uint8_t magic;
+    uint8_t check;
+    uint16_t size;
+} __attribute__((packed));
```

Comment:
Is this a final solution? 8-bit magic number and 8-bit check based on length which is 1500B maximum does not provide enough combinations to be reliable enough, in my opinion.

Reply:
@kostaond Per my calculations, it's about 1 in 11,400 packets of random data that could be wrongly identified as a valid packet header.

**Long story**
* P of finding magic in random data: `P(magic)=1/256`
* P of checksum: `P(check)=1501/65536`
* combined: `8.8×10^-5`, which corresponds to the above number

So I think we're more than safe, here.

PS: I'd still apply the XOR idea above, but will do it in the upcoming fix

Reply:
Yeah, XOR idea is definitely better than just simple sum as was used before because it should ensure more even distribution.